include/aspect/simulator.h

```diff
@@ -2104,6 +2104,13 @@ namespace aspect
       template <int dimension, int velocity_degree>
       friend class StokesMatrixFreeHandlerImplementation;
       friend struct Parameters<dim>;
+
+      /** Parameters for the adjoint problem
+       */
+      LinearAlgebra::BlockVector                                current_adjoint_solution;
+      void solve_stokes_adjoint ();
+      void compute_parameter_update ();
```

Comment:
I've renamed this to 'compute_sensitivity_kernel' because that's really what it does and then stores that in the compositional field. The material model then updates the parameters based on that. I've also improved the documentation here. Let me know if you think this name is okay or should be changed.